src/internal/helper.ts

```diff
@@ -0,0 +1,21 @@
+type OptionalKeys<T> = { [K in keyof T]: T[K] extends Required<T>[K] ? never : K }[keyof T];
+type InjectorFor<T, K extends keyof T> = Exclude<keyof T, K | OptionalKeys<T>> extends never
+    ? { [P in K]: Promise<T[P]> }
+    : never;
+
+export async function pick<T, K extends keyof T>(
+    injector: { [P in K]: Promise<T[P]> },
+    keys: K[],
+): Promise<{ [P in K]: T[P] }> {
```

Comment:
How about this?

```
async function pick<T, K extends keyof T>(
  injector: { [P in K]: Promise<T[P]> },
  keys: readonly K[]
): Promise<{ [P in K & typeof keys[number]]: T[P] }> {
    return Object.fromEntries(await Promise.all(keys.map(async name => [name, await injector[name]])));
}
```

- `K[]` -> `readonly K[]` because no need to make it modifyable
- `[P in K & typeof keys[number]]` because resulted object *may* not all properties that requested with `K` type
  - When `keys` array has *subset of* the `K`, the resulted object only contains properties that listed in the `keys` argument.

Reply:
![image](https://user-images.githubusercontent.com/2661835/89143540-e6eed580-d585-11ea-91bd-22bbcaf33142.png)
![image](https://user-images.githubusercontent.com/2661835/89143678-641a4a80-d586-11ea-814f-4493f1e8a5ed.png)

The issue you mentioned exists, but this change doesn't seem to correct the issue. Maybe there is something I'm missing?

Reply:
Oops, I wrote this code based on my experience (different code base). But as you mentioned, this code does not solve the problem...
I forget how to deal with it in detail...

Do you have any good idea to detect such problem?

Reply:
Hmm, I don't come up with an idea to fix this issue.

Maybe we just give up exposing `pick` since `inject` covers most use cases we need.

Reply:
> Maybe we just give up exposing pick since inject covers most use cases we need.

Agree!
It is possible to expose `inject` when we find good solution.